nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -642,6 +641,11 @@ in
       }
     ];
 
+    systemd.tmpfiles.rules = [
+      "d ${cfg.stateDir} 0750 ${cfg.user} ${cfg.group} - -"
+      "d /run/nginx 0750 ${cfg.user} ${cfg.group} - -"
```

Comment:
In the case of a hard code path like /run/nginx instead of tmpfiles.d you should probably use:
`serviceConfig.RuntimeDirectory = "nginx";
serviceConfig.RuntimeDirectoryMode = "0750";`

Reply:
Thanks, update.